ARCHITECTURE.md

```diff
@@ -0,0 +1,97 @@
+# OMF mds-core Architecture
+
+## Draft
+# Background
+The mds-core repo represents an implementation of the MDS Agency and Policy APIs, as well as some not-yet-ratified API implementations (Compliance, Audit, Geography, Jurisdictions), plus shared libraries, containerization, authentication, etc.  
+
+The goal of this document is to give an overview of the technology choices, design, and implementation of these services.   The mds-core repo will be a reference system that can be used as a basis for building and testing additional services, as well as a reasonable facsimile of what is running for the first city to deploy these services (Los Angeles).
+# Languages
+This implementation is built in [TypeScript](https://www.typescriptlang.org) v3.8.  The first draft of the Agency service was built in vanilla EMCAScript 6, and types were added as the system grew.
+
+TypeScript has a large developer community, a comprehensive set of toolchains, wide support in cloud providers, and is reasonably performant.  JavaScript is the [most popular](https://octoverse.github.com/) language inside GitHub, with TypeScript 7th.
+
+The mds-core endpoints are implemented using the popular framework [express.js](https://expressjs.com/).
+
+# Authentication & Identity
+We authenticate and authorize access to APIs and public-facing websites using JSON Web Tokens (JWT). JWTs are issued by Auth0 using the OAuth 2.0 protocol. 
+## Auth0
```

Comment:
Is this only meant to be an explanation of how our implementation works? I think a sentence or two about whether we recommend using Auth0 and the pros of cons of other authentication alternatives might be nice too.

Reply:
Yeah, I agree that adding some additional information about Auth0 would be great